src/java/org/apache/cassandra/index/sai/disk/v1/MetadataWriter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.v1;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.cassandra.index.sai.disk.io.RAMIndexOutput;
+import org.apache.cassandra.index.sai.utils.SAICodecUtils;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.util.BytesRef;
+
+@NotThreadSafe
+public class MetadataWriter implements Closeable
+{
+    private final IndexOutput output;
+    private final Map<String, BytesRef> map = new HashMap<>();
+
+    public MetadataWriter(IndexOutput output)
+    {
+        this.output = output;
+    }
+
+    public Builder builder(String name)
+    {
+        return new Builder(name);
+    }
+
+    public class Builder extends RAMIndexOutput implements Closeable
+    {
+        private Builder(String name)
+        {
+            super(name);
+        }
+
+        @Override
+        public void close()
+        {
+            map.put(getName(), new BytesRef(out.getBytes(), 0, out.getPosition()));
```

Comment:
nit: Wondering if `out` should just be private, w/ `getBytes()` exposing `out.getBytes()` and `getFilePointer()` used for the third argument here.

Reply:
If I understand this correctly, I've added a `getBytes` to `RAMIndexOutput` to return a `BytesRef` and have made `out` private in that class.